lib/src/base_response.dart

```diff
@@ -2,6 +2,8 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:io';
```

Comment:
We can't import `dart:io` from this library. `universal_io` is intending to get around that problem, but we don't want to take a dependency on that library either. If we wanted to have this functionality in this library we'd have to reimplement an interface for `Cookie`.

This feels like the type of thing that should be added on in a side package (which could more easily have a dependency on `universal_io`)

```dart
extension Cookies on Response {
  List<Cookie> get cookies {
    // ...
  }
}
```

I suppose the nice thing about implementing it here is being able to cache it, but using an `expando` is probably a fine alternative if caching is important. Probably you'll only read `.cookies` once anyway.

Reply:
Thanks for your review @natebosch !

I found several issues have been raised regarding this case of multiple `set-cookie`s, and I feel that a fundamental solution is needed. At the same time, I also understood the difficulties related to `dart:io`. It's quite a deep-rooted problem.

It's possible to add a `Cookie` interface and implementation, but is there a reluctance to add a `Cookie` object as a development policy for the http library?

If we can implement `Cookie` interface in http library, I thought the following example was a very good solution.

```dart
extension Cookies on Response {
  List<Cookie> get cookies {
    // ...
  }
}
```